src/client/profile-provider.ts

```diff
@@ -417,6 +421,63 @@ export class ProfileProvider {
     return result;
   }
 
+  private async cacheProviderInfo(providerName: string): Promise<ProviderJson> {
+    const errors: Error[] = [];
+    if (this.providerJson === undefined) {
+      const cachePath = joinPath(Config.instance().cachePath, 'providers');
+      const providerCachePath = joinPath(cachePath, `${providerName}.json`);
+
+      // If we don't have provider info, we first try to fetch it from the registry
+      try {
+        this.providerJson = await fetchProviderInfo(providerName);
+        try {
+          await fsp.mkdir(cachePath, { recursive: true });
+          await fsp.writeFile(
+            providerCachePath,
+            JSON.stringify(this.providerJson)
```

Comment:
Nitpick: nice formatting? (JSON.stringify(this.providerJson, undefined,2))

Reply:
it's not really meant to be read :thinking:

Reply:
Yeah, I know. I was just an idea. It would be nice for people who decide to dive deeper to superface.